packages/front-end/components/Settings/RestoreConfigYamlButton.tsx

```diff
@@ -17,6 +17,8 @@ import cloneDeep from "lodash/cloneDeep";
 import UploadConfigYml from "./UploadConfigYml";
 
 function sanitizeSecrets(d: DataSourceInterfaceWithParams) {
+  if (!d || !d.params) return;
```

Comment:
This fixes that one specific error, but I'm sure there are countless others that may be caused by different typos.  Can we wrap the entire parsing process in a try/catch so it fails gracefully?  Right now on production, it completely breaks the app and shows a blank white screen.

Reply:
Makes sense to me, I will make that happen.